src/snowflake/snowpark/_internal/data_source/drivers/psycopg2_driver.py

```diff
@@ -0,0 +1,362 @@
+    # ANYELEMENTOID = 2283  # Not listed in the Postgres doc.
+    UUID = 2950  # Not listed in the pgtypes.h
+    TXID_SNAPSHOT = 2970  # Not listed in the pgtypes.h
+    PG_LSN = 3220  # Not listed in the pgtypes.h
+    TSVECTOR = 3614  # Not listed in the pgtypes.h
+    TSQUERY = 3615  # Not listed in the pgtypes.h
+    JSONB = 3802  # Not listed in the pgtypes.h
+    PG_SNAPSHOT = 5038  # Not listed in the pgtypes.h
+
+
+# https://other-docs.snowflake.com/en/connectors/postgres6/view-data#postgresql-to-snowflake-data-type-mapping
+BASE_POSTGRES_TYPE_TO_SNOW_TYPE = {
+    Psycopg2TypeCode.BOOLOID: BooleanType,
+    Psycopg2TypeCode.BYTEAOID: BinaryType,
+    Psycopg2TypeCode.CHAROID: StringType,
+    Psycopg2TypeCode.INT8OID: IntegerType,
+    Psycopg2TypeCode.INT2OID: IntegerType,
+    Psycopg2TypeCode.INT4OID: IntegerType,
+    Psycopg2TypeCode.TEXTOID: StringType,
+    Psycopg2TypeCode.POINTOID: StringType,
+    Psycopg2TypeCode.LSEGOID: StringType,
+    Psycopg2TypeCode.PATHOID: StringType,
+    Psycopg2TypeCode.BOXOID: StringType,
+    Psycopg2TypeCode.POLYGONOID: StringType,
+    Psycopg2TypeCode.LINEOID: StringType,
+    Psycopg2TypeCode.FLOAT4OID: FloatType,
+    Psycopg2TypeCode.FLOAT8OID: DoubleType,
+    Psycopg2TypeCode.CIRCLEOID: StringType,
+    Psycopg2TypeCode.CASHOID: VariantType,
+    Psycopg2TypeCode.MACADDROID: StringType,
+    Psycopg2TypeCode.CIDROID: StringType,
+    Psycopg2TypeCode.INETOID: StringType,
+    Psycopg2TypeCode.BPCHAROID: StringType,
+    Psycopg2TypeCode.VARCHAROID: StringType,
+    Psycopg2TypeCode.DATEOID: DateType,
+    Psycopg2TypeCode.TIMEOID: TimeType,
+    Psycopg2TypeCode.TIMESTAMPOID: TimestampType,
+    Psycopg2TypeCode.TIMESTAMPTZOID: TimestampType,
+    Psycopg2TypeCode.INTERVALOID: StringType,
+    Psycopg2TypeCode.TIMETZOID: TimeType,
+    Psycopg2TypeCode.BITOID: StringType,
+    Psycopg2TypeCode.VARBITOID: StringType,
+    Psycopg2TypeCode.NUMERICOID: DecimalType,
+    Psycopg2TypeCode.JSON: VariantType,
+    Psycopg2TypeCode.JSONB: VariantType,
+    Psycopg2TypeCode.MACADDR8: StringType,
+    Psycopg2TypeCode.UUID: StringType,
+    Psycopg2TypeCode.XML: StringType,
+    Psycopg2TypeCode.TSVECTOR: StringType,
+    Psycopg2TypeCode.TSQUERY: StringType,
+    Psycopg2TypeCode.TXID_SNAPSHOT: StringType,
+    Psycopg2TypeCode.PG_LSN: StringType,
+    Psycopg2TypeCode.PG_SNAPSHOT: StringType,
+}
+
+
+class Psycopg2Driver(BaseDriver):
+    def __init__(
+        self, create_connection: Callable[[], "Connection"], dbms_type: Enum
+    ) -> None:
+        super().__init__(create_connection, dbms_type)
+
+    def to_snow_type(self, schema: List[Any]) -> StructType:
+        # The psycopg2 spec is defined in the following links:
+        # https://www.psycopg.org/docs/cursor.html#cursor.description
+        # https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.Column
+        fields = []
+        for (
+            name,
+            type_code,
+            _display_size,
```

Comment:
nit: what is the reason we use '_' here?

Reply:
this is pylint suggestion for unused var